### PR TITLE
Fixes#1048 cisco xr admin show inventory

### LINKED ISSUE #1048: cisco_xr_admin_show_inventory no parse correctly in several models
##### ISSUE TYPE
 - Bug Report
 
##### TEMPLATE USING
 cisco_xr_admin_show_inventory.textfsm

Value NAME (.*)
Value DESCR (.*)
Value PID (([\S+]+|.*))
Value VID (.*)
Value SN ([\w+\d+]+)

Start
  ^NAME:\s+"${NAME}",\s+DESCR:\s+"${DESCR}"
  ^PID:\s+${PID}.*,.*VID:\s+${VID},.*SN:\s+${SN} -> Record
  ^PID:\s+,.*VID:\s+${VID},.*SN: -> Record
  ^PID:\s+${PID}.*,.*VID:\s+${VID},.*SN: -> Record
  ^PID:\s+,.*VID:\s+${VID},.*SN:\s+${SN} -> Record
  ^PID:\s+${PID}.*,.*VID:\s+${VID}.*
  ^PID:\s+,.*VID:\s+${VID}.*
  ^.*SN:\s+${SN} -> Record
  ^.*SN: -> Record

##### SAMPLE COMMAND OUTPUT
[{'name': '', 'descr': '', 'pid': '', 'vid': '', 'sn': 'FOC23XXXXXX'},
 {'name': '', 'descr': '', 'pid': '', 'vid': '', 'sn': 'POXXXXXXXXX'},
 {'name': '', 'descr': '', 'pid': '', 'vid': '', 'sn': 'POXXXXXXXXX'}]

##### SUMMARY
The template works correctly in models:

ASR 9006 4 Line Card Slot Chassis

The template DOES NOT work correctly in models:

ASR 9901 Chassis version 6.6.3
ASR 9906 Chassis version 6.7.3

In these models there is a white space at the beginning of the lines, so DO NOT start with ^NAME or ^PID

##### STEPS TO REPRODUCE
Output from ASR 9901 and 9906:
```
#admin show inventory
Fri Dec 24 13:42:23.980 CET

 Name: Rack 0                Descr: ASR 9906 4 Line Card Slot Chassis
 PID: ASR-9906               VID: V01                   SN: FOXXXXXXXX

 Name: 0/0                   Descr: 24X10G/1G Packet Transport Optimized LC
 PID: A9K-24X10GE-1G-TR      VID: V01                   SN: FOCYYYYYYYY

 Name: 0/1                   Descr: 48X10G/1G Packet Transport Optimized LC
 PID: A9K-48X10GE-1G-TR      VID: V01                   SN: FOC2XXXXXX
```

NOTE: i hope the white space before NAME and PID was clear

### PR DESCRIPTION
**Pull request by diepes**
##### ISSUE TYPE
 - Bugfix Pull Request
 - Additional Testing

##### COMPONENT
cisco_xr  admin show inventory

##### SUMMARY
Fixes #1048
1. Add new raw test tests/cisco_xr/admin_show_inventory/cisco_xr_admin_show_inventory2.yml
   *  update parsing

3. Small fix on existing template, remove space from match.
   * e.g. tests/cisco_xr/admin_show_inventory/cisco_xr_admin_show_inventory.yml
      `vid: "V01 "`  ->     `vid: "V01"`

### CODE REVIEW

File: ntc_templates/templates/cisco_xr_admin_show_inventory.textfsm [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-  ^.*SN:\s+${SN} -> Record
-  ^.*SN: -> Record
-
+  ^NAME:\s+"${NAME}",\s+DESCR:\s+"${DESCR}"\s*$$
+  ^\s*Name:\s+${NAME}\s+Descr:\s+${DESCR}\s*$$
+  ^\s*PID:\s+${PID}\s*,\s+VID:\s+${VID}\s*,\s+SN:\s*${SN}$$ -> Record
+  ^\s*PID:\s+${PID}\s+VID:\s+${VID}\s+SN:\s+${SN}$$ -> Record
+  #
+  ^\s*$$ 
+  ^(Mon|Tue|Wed|Thu|Fri|Sat|Sun)\s\S{3}\s+\d
```
Comment (jvanderaa) [inline ntc_templates/templates/cisco_xr_admin_show_inventory.textfsm:14]: I haven't looked at an XR in some time. Does this output also happen in a programatic interface with Netmiko/Napalm?

File: tests/cisco_xr/admin_show_inventory/cisco_xr_admin_show_inventory2.raw [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Fri Dec 24 13:42:23.980 CET
+
+ Name: Rack 0                Descr: ASR 9906 4 Line Card Slot Chassis
+ PID: ASR-9906               VID: V01                   SN: FOXXXXXXXX
+
+ Name: 0/0                   Descr: 24X10G/1G Packet Transport Optimized LC
+ PID: A9K-24X10GE-1G-TR      VID: V01                   SN: FOCYYYYYYYY
+
+ Name: 0/1                   Descr: 48X10G/1G Packet Transport Optimized LC
+ PID: A9K-48X10GE-1G-TR      VID: V01                   SN: FOC2XXXXXX
```
Comment (jmcgill298) [inline tests/cisco_xr/admin_show_inventory/cisco_xr_admin_show_inventory2.raw:10]:
```suggestion
 PID: A9K-48X10GE-1G-TR      VID: V01                   SN: FOC2XXXXXX
 
```

**Review by jmcgill298 — CHANGES_REQUESTED** [review]

**Comment by andyb2000** [comment]
Hi, any movement on this?
I've tried to apply to mine for testing and it fails parsing. Example output from my "admin show inventory":

 Name: Rack 0                Descr: ASR-9010 Chassis
 PID: ASR-9010               VID: V01                   SN: FOX0000X0XX

 Name: 0/0                   Descr: 24X10G/1G  Service Edge Optimized LC
 PID: A9K-24X10GE-1G-SE      VID: V05                   SN: FOC0000XXXX

 Name: 0/7                   Descr: ASR 9000 8-port 100GE FLEXE SE linecard
 PID: A9K-8HG-FLEX-SE        VID: V02                   SN: FOC0000XXXX

And parsing fails:
Parsing errors
('State Error raised. Rule Line: 15. Input Line:  PID: ASR-9010               '
 'VID: V01                   SN: FOX0000X0XX')

(Differences to previous seem to be lacking quote marks and the spacing)

**Comment by diepes** [comment]
@andyb2000 added you example as additional test.

**Pull request merged**